src/request.c

```diff
@@ -173,6 +188,12 @@ request_unexport (Request *request)
   g_object_unref (request);
 }
 
+void
+request_sender_died (Request *request)
+{
+  g_signal_emit (request, signals[SENDER_DIED], 0);
```

Comment:
This doesn't look threadsafe at all. The AppIdData doesn't ref the requests, so it may very well be dead by now. And in general emitting signals cross-threads is fraught with issues.